src/OpStateBridge.sol

```diff
@@ -0,0 +1,205 @@
+    /// @param _opGasLimit The new opGasLimit for transferOwnershipOptimism
+    event SetGasLimitTransferOwnershipOp(uint32 _opGasLimit);
+
+    ///////////////////////////////////////////////////////////////////
+    ///                            ERRORS                           ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice Thrown when an attempt is made to renounce ownership.
+    error CannotRenounceOwnership();
+
+    ///////////////////////////////////////////////////////////////////
+    ///                         CONSTRUCTOR                         ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice constructor
+    /// @param _worldIDIdentityManager Deployment address of the WorldID Identity Manager contract
+    /// @param _opWorldIDAddress Address of the Optimism contract that will receive the new root and timestamp
+    /// @param _crossDomainMessenger L1CrossDomainMessenger contract used to communicate with the desired OP
+    /// Stack network
+    constructor(
+        address _worldIDIdentityManager,
+        address _opWorldIDAddress,
+        address _crossDomainMessenger
+    ) {
+        opWorldIDAddress = _opWorldIDAddress;
+        worldIDAddress = _worldIDIdentityManager;
+        crossDomainMessengerAddress = _crossDomainMessenger;
+        _gasLimitSendRootOp = 100000;
+        _gasLimitSetRootHistoryExpiryOp = 100000;
+        _gasLimitTransferOwnershipOp = 100000;
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    ///                          PUBLIC API                         ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice Sends the latest WorldID Identity Manager root to the IOpStack.
+    /// @dev Calls this method on the L1 Proxy contract to relay roots to the destination OP Stack chain
+    function propagateRoot() external {
+        uint256 latestRoot = IWorldIDIdentityManager(worldIDAddress).latestRoot();
+
+        // The `encodeCall` function is strongly typed, so this checks that we are passing the
+        // correct data to the optimism bridge.
+        bytes memory message = abi.encodeCall(IOpWorldID.receiveRoot, (latestRoot));
+
+        ICrossDomainMessenger(crossDomainMessengerAddress).sendMessage(
+            // Contract address on the OP Stack Chain
+            opWorldIDAddress,
+            message,
+            _gasLimitSendRootOp
+        );
+
+        emit RootPropagated(latestRoot);
+    }
+
+    /// @notice Adds functionality to the StateBridge to transfer ownership
+    /// of OpWorldID to another contract on L1 or to a local OP Stack chain EOA
+    /// @param _owner new owner (EOA or contract)
+    /// @param _isLocal true if new owner is on Optimism, false if it is a cross-domain owner
+    function transferOwnershipOp(address _owner, bool _isLocal) external onlyOwner {
```

Comment:
Do we need the `Op` suffix here?

Reply:
In this case yet, so it doesn't collide with `Ownable`'s `transferOwnership`